src/TrackedFloat.jl

```diff
@@ -25,95 +25,114 @@ end
+# floatmin2(::Type{TrackedFloat64}) = reinterpret(Float64, 0x21a0000000000000)
+
 for TrackedFloatN in (:TrackedFloat16, :TrackedFloat32, :TrackedFloat64)
 
-# FloatN is the base float type derived from TrackedFloatN
-@eval FloatN = $(Symbol("Float", string(TrackedFloatN)[end-1:end]))
+  # FloatN is the base float type derived from TrackedFloatN
+  @eval FloatN = $(Symbol("Float", string(TrackedFloatN)[end-1:end]))
 
-@eval begin
-  struct $TrackedFloatN <: AbstractTrackedFloat
-    val::$FloatN
-  end
+  @eval begin
+    struct $TrackedFloatN <: AbstractTrackedFloat
+      val::$FloatN
+    end
 
-  Base.Float64(x::$TrackedFloatN) = Float64(x.val)
-  Base.Float32(x::$TrackedFloatN) = Float32(x.val)
-  Base.Float16(x::$TrackedFloatN) = Float16(x.val)
-  Base.Int64(x::$TrackedFloatN) = Int64(x.val)
-  Base.Int32(x::$TrackedFloatN) = Int32(x.val)
-  Base.Int16(x::$TrackedFloatN) = Int16(x.val)
-
-  Base.bitstring(x::$TrackedFloatN) = bitstring(x.val)
-  Base.show(io::IO,x::$TrackedFloatN) = print(io, $TrackedFloatN,"(",string(x.val),")")
-
-  # $TrackedFloatN(x::AbstractFloat) = $TrackedFloatN(x)
-  # $TrackedFloatN(x::Integer) = $TrackedFloatN(x)
-  $TrackedFloatN(x::Rational{}) = $TrackedFloatN($FloatN(x))
-  $TrackedFloatN(x::$TrackedFloatN) = $TrackedFloatN(x.val)
-  $TrackedFloatN(x::Bool) = $TrackedFloatN($FloatN(x))
-
-  Base.promote_rule(::Type{<:Integer},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Float64},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Float32},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Float16},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Bool},::Type{$TrackedFloatN}) = $TrackedFloatN
-end
+    Base.Float64(x::$TrackedFloatN) = Float64(x.val)
+    Base.Float32(x::$TrackedFloatN) = Float32(x.val)
+    Base.Float16(x::$TrackedFloatN) = Float16(x.val)
+    Base.Int64(x::$TrackedFloatN) = Int64(x.val)
+    Base.Int32(x::$TrackedFloatN) = Int32(x.val)
+    Base.Int16(x::$TrackedFloatN) = Int16(x.val)
+
+    Base.bitstring(x::$TrackedFloatN) = bitstring(x.val)
+    Base.show(io::IO,x::$TrackedFloatN) = print(io, $TrackedFloatN,"(",string(x.val),")")
+
+    # $TrackedFloatN(x::AbstractFloat) = $TrackedFloatN(x)
+    # $TrackedFloatN(x::Integer) = $TrackedFloatN(x)
+    $TrackedFloatN(x::Rational{}) = $TrackedFloatN($FloatN(x))
+    $TrackedFloatN(x::$TrackedFloatN) = $TrackedFloatN(x.val)
+    $TrackedFloatN(x::Bool) = $TrackedFloatN($FloatN(x))
+
+    Base.promote_rule(::Type{<:Integer},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Float64},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Float32},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Float16},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Bool},::Type{$TrackedFloatN}) = $TrackedFloatN
+
+    floatmin2($TrackedFloatN) = floatmin2($FloatN)
```

Comment:
Does this `floatmin2` work for tiny examples?

It looks strange to me. I'd expect something like `Base.Int64` above, where there's a name for the argument that gets used in the function body. I see there's something going on below to define `floatmin2` but that also doesn't seem to use its argument.

Reply:
Ohhh I see. `floatmin2` expects a type and returns a float. We should have it return something like `reinterpret(TrackedFloat64, 0x21a0000000000000)`

Reply:
Better yet: `floatmin2(::TrackedFloat64) = TrackedFloat64(floatmin2(Float64))`